src/libos/src/fs/syscalls.rs

```diff
@@ -35,6 +37,54 @@ pub fn do_eventfd2(init_val: u32, flags: i32) -> Result<isize> {
     Ok(fd as isize)
 }
 
+pub fn do_timerfd_create(clockid: u32, flags: i32) -> Result<isize> {
+    info!("timerfd: clockid {}, flags {} ", clockid, flags);
+
+    let inner_flags =
+        TimerCreationFlags::from_bits(flags).ok_or_else(|| errno!(EINVAL, "invalid flags"))?;
+    let file_ref: Arc<dyn File> = {
+        let timer = TimerFile::new(clockid, inner_flags)?;
+        Arc::new(timer)
+    };
+
+    let fd = current!().add_file(
+        file_ref,
+        inner_flags.contains(TimerCreationFlags::TFD_CLOEXEC),
+    );
+    Ok(fd as isize)
+}
+
+pub fn do_timerfd_settime(
+    fd: FileDesc,
+    flags: u32,
+    new_value_ptr: *const itimerspec_t,
+    old_value_ptr: *mut itimerspec_t,
+) -> Result<isize> {
+    let current = current!();
+    let file = current.file(fd)?;
+    let timerfile = file.as_timer()?;
+    from_user::check_ptr(new_value_ptr)?;
+    let new_value = unsafe { new_value_ptr.read() };
+    let old_value = timerfile.settime(flags, &new_value)?;
```

Comment:
set_time

Reply:
After a deep review of timerfd_settime, we should check the `flags` before calling `timerfile.set_time`